CMakeLists.txt

```diff
@@ -238,24 +235,59 @@ if (enable_checksums)
   set(SRC ${SRC} checksums.c)
 endif(enable_checksums)
 
-add_library(gc-lib STATIC ${SRC})
-set_target_properties(gc-lib PROPERTIES
-                      COMPILE_DEFINITIONS GC_NOT_DLL)
-#TODO TARGET_LINK_LIBRARIES(...  ...  ${LIBS})
-
-add_library(gcmt-lib STATIC ${SRC})
-set_target_properties(gcmt-lib PROPERTIES
-                      COMPILE_DEFINITIONS GC_NOT_DLL)
+# Create target
+add_library(gc ${SRC})
+target_include_directories(gc  
+    PUBLIC     
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:include>
+)
+if (NOT BUILD_SHARED_LIBS)
+    target_compile_definitions(gc PRIVATE GC_NOT_DLL)
+else()
+    target_compile_definitions(gc PRIVATE GC_DLL)
+endif()
 
-add_library(gcmt-dll SHARED ${SRC})
+# Install library
+install(
+    TARGETS gc
+    LIBRARY DESTINATION lib 
+    ARCHIVE DESTINATION lib
+    RUNTIME DESTINATION bin 
+    INCLUDES DESTINATION include
+)
+
+# Install headers
+if (NOT BDWGC_SKIP_HEADERS)
+    file(GLOB BDWGC_HEADERS "include/*.h")
+    install(FILES ${BDWGC_HEADERS} DESTINATION include/gc)
+
+    file(GLOB BDWGC_EXTRA_HEADERS "include/extra/*.h")
+    install(FILES ${BDWGC_EXTRA_HEADERS} DESTINATION include)
+endif()
 
-if (WIN32)
-  add_executable(cord cord/cordbscs.c cord/cordxtra.c
-                 cord/tests/de.c cord/tests/de_win.c)
-  set_target_properties(cord PROPERTIES WIN32_EXECUTABLE TRUE)
-  set_target_properties(cord PROPERTIES COMPILE_DEFINITIONS GC_NOT_DLL)
-  target_link_libraries(cord gc-lib)
-  target_link_libraries(cord gdi32)
+# Install tools
+if (WIN32 AND BDWGC_INSTALL_CORD)
+    add_executable(cord 
+                   cord/cordbscs.c 
+                   cord/cordxtra.c
+                   cord/tests/de.c
+                   cord/tests/de_win.c)
+    set_target_properties(cord PROPERTIES WIN32_EXECUTABLE TRUE)
+    target_compile_definitions(cord PRIVATE GC_NOT_DLL)
+
+    target_link_libraries(cord PRIVATE gc gdi32)
+    target_include_directories(cord
+        PRIVATE      
+            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    )
+
+    install(
+        TARGETS cord 
+        DESTINATION tools/cord 
+    )
 endif()
 
-add_subdirectory(tests)
+if (NOT BDWGC_SKIP_TESTS)
```

Comment:
I added an option to control whether tests are installed here.
It is ON by default.

Reply:
Having an option is OK.
Test generation should be on by default. Is it OK?